kafka/common/pkg/kafka/consumer_factory.go

```diff
@@ -23,19 +23,18 @@ import (
 	"go.uber.org/zap"
 )
 
-var newConsumerGroupFromClient = sarama.NewConsumerGroupFromClient
-
 // Kafka consumer factory creates the ConsumerGroup and start consuming the specified topic
 type KafkaConsumerGroupFactory interface {
 	StartConsumerGroup(groupID string, topics []string, logger *zap.Logger, handler KafkaConsumerHandler) (sarama.ConsumerGroup, error)
 }
 
 type kafkaConsumerGroupFactoryImpl struct {
-	client sarama.Client
+	config               *sarama.Config
+	addrs                []string
+	newConsumerGroupFunc func(addrs []string, groupID string, config *sarama.Config) (sarama.ConsumerGroup, error)
```

Comment:
Do we really need this function here? Can we continue to use the `newConsumerGroupFromClient` like before?

Reply:
I thought it was better not to have a global variable.

Reply:
It's weird in both cases, but the global var maybe it's better since this func is not data related to the consumer factory :smile:

Reply:
updated